app.py

```diff
@@ -1,39 +1,77 @@
-from lego_prediction import pricer
+from flask import Flask, render_template, request, redirect
+from flask_sqlalchemy import SQLAlchemy
+from form import UrlForm
+import shortuuid
 from werkzeug import exceptions
+import os 
+
 
 app = Flask(__name__)
-CORS(app)
+app.config['SECRET_KEY'] = '5204956a60384b5685e8ce01e34235517b576fc6a461ff13'
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///site.db'
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+db = SQLAlchemy(app)
+
+@app.before_first_request
+def create_tables():
+    db.create_all()
+class UrlModel(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    url = db.Column(db.String(250), unique=True, nullable=False)
+    url_short = db.Column(db.String(100), nullable=False)
+
+    def __repr__(self):
+        return f"UrlModel('{self.url}', '{self.url_short}')"
+
+def find_url(url):
+    found = UrlModel.query.filter_by(url=url).first()
+    print(found)
+    return found
 
-from flask_mail import Mail
-from flask_mail import Message
-mail = Mail(app)
+
+def find_short(short):
+    found = UrlModel.query.filter_by(url_short=short).first()
+    print(found)
+    return found
+
+
+def isValid(id):
+    if len(id) != 7:
+        return False
+    else:
+        return True
+
+def init_db():
+    db.create_all()
 
 @app.route('/')
 def home():
     return render_template('home.html', title='home')
 
-@app.route('/prediction', methods=['GET', 'POST'])
-def predict():
+@app.route("/short", methods=['GET', 'POST'])
+def short():
+    form = UrlForm()
     if request.method == 'POST':
-        print(request.form)
-        piece_count = request.form['piece_count']
-        result = pricer.predict(piece_count)
-        return render_template('predict.html', default=piece_count, result=result)
+        if find_url(form.url.data):
+            found_url = find_url(form.url.data).url
+            found_short = find_url(form.url.data).url_short
+            return render_template('result.html', url=found_url, url_short=found_short)
+        else:
+            short = shortuuid.uuid()[:7]
+            url = UrlModel(url=form.url.data, url_short=short)
+            db.session.add(url)
+            db.session.commit()
+            return render_template('result.html', url=url.url, url_short=short)
     else:
-        return render_template('predict.html', default=0, result=0, title='Predict')
-
-@app.route('/reminder', methods=['GET', 'POST'])
-def reminder():
-    piece_count = request.form['piece_count']
-    result = request.form['result']
-    to_email = request.form['to_email']
-    msg = Message("Your LegOh! prediction is here!", sender="futureproof.testing@gmail.com", recipients=[to_email])
-    msg.html = render_template('mailers/your_result.html', piece_count=piece_count, result=result)
-    # mail.send(msg)
-    return render_template('thankyou.html')
+        return render_template('short.html', form=form)
 
+@app.route("/<string:id>")
+def change(id):
+    if isValid(id):
```

Comment:
Able to break the site by entering http://localhost:5000/1234567
Add error handling if url isnt in database